content/blog/nodepool-upgrades/index.md

```diff
@@ -0,0 +1,161 @@
+
+::: div image-l
+<svg title='Animation showing an in-place upgrade for the nodes in a Kubernetes cluster [2].' src='in-place-upgrade.gif' />
+:::
+
+The advantage of an in-place upgrade is that it requires minimal additional compute resources (a single additional node). The downside to this strategy is that it can take quite a bit of time as nodes are drained and upgraded one at a time in series. Additionally, pods may need to make more than 1 move as they are shuffled around during the draining of nodes.
+
+For an **out-of-place upgrade**, a fresh node pool is created with the new Kubernetes version. Once the new nodes are all running, you can cordon the old node pool, drain the old nodes one by one, and then delete the old node pool. This strategy is visualized in the animation below:
+
+::: div image-l
+<svg title='Animation showing an out-of-place upgrade for the nodes in a Kubernetes cluster (diagram from Fairwinds).' src='out-of-place-upgrade.gif' />
+:::
+
+An out-of-place upgrade requires a temporary doubling of compute resources in exchange for a shorter upgrade window. The decrease in upgrade duration results from the parallelization of the startup time of the new upgraded nodes, as well as the minimization of the movement of the pods. In this strategy, pods make a single move from the old node to the new upgraded node.
+
+Assuming you’re okay with the temporary increase in compute resource utilization, we recommend utilizing the out-of-place upgrade strategy to decrease the amount of time you’ll need to be available to babysit the upgrade process.
+
+## Configuring K8s Resources
+
+Whichever worker node upgrade strategy you choose will involve shuffling of your pods from the original nodes to an upgraded node. If your resources are not properly configured, this can cause downtime. Let’s take a look at a few potential pitfalls.
+
+### Standalone Pods
+
+A pod is the smallest deployable object in Kubernetes. It represents a single instance of an application running in your cluster. Pods are ephemeral; if a pod is evicted from a node, the pod does not replace itself. Because pods are not self-healing, it is not recommended that you create individual Pods directly. Instead, use a controller such as a Deployment to create and manage the Pod for you.
+
+To minimize downtime, ensure that all your pods are managed by a ReplicaSet, Deployment, StatefulSet or something similar. Standalone pods might need to be manually rescheduled after an upgrade.
+
+### Daemonsets
+
+A [DaemonSet](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/) ensures that all (or some) nodes run a copy of a pod. Daemonsets are often used for node monitoring or log collection and do not typically serve traffic. For these use cases, it’s usually acceptable to have a small gap in data during the worker node upgrade.
+
+### Deployments
```

Comment:
I would put this one before Daemonsets

Reply:
I'll move it to the bottom so that Deployments and StatefulSets can be next to each other.